### DISCUSSION TITLE
add_freetext_annot: Use CJK fontname/ Different fontname within one annot

### DISCUSSION THREAD

**Discussion by sldzys** [pymupdf/PyMuPDF#1109] [Looking for help] [answered]
Hello,

Thanks for this amazing library! 

1. I am attempting to insert a 'text box' annotation to PDF with the fontname of 'china-ss'. But i failed. This is my code：
```
import fitz
doc = fitz.open("./test.pdf")         
page = doc[0]                        
rect = fitz.Rect(38, 65, 130, 90)  
annot = page.add_freetext_annot(
        rect,
        '你好',
        fontsize=12,
        rotate=0,
        fontname="china-ss",
        text_color=(0, 0, 1),
        fill_color=(1, 1, 0),
        align=fitz.TEXT_ALIGN_LEFT
        )
annot.set_border(width=0.5, dashes=[2])
annot.update(text_color=(0, 0, 1), fill_color=(1, 1, 0),border_color=(0, 1, 0))
doc.save("./test_anno.pdf")  
```

It seems that running with ‘insertText’ can display the font normally but I need to generate an annotation in pdf that can be modified/moved with adobe software.

2. Is it possible to use multiple font names in the same text box which can be modified or moved in Adobe app? For example, use 'china-ss' for Chinese characters and 'Arial' for others. This seems to be possible in ‘fillTextbox’, like the reference link below. But this method seems to embed the annotation in the pdf file instead of a pdf object of annotated.(cannot modified or moved in Adobe)

[https://github.com/pymupdf/PyMuPDF/issues/549](url)

I'm wondering if I have missed anything?

My configration
- Win 10 Pro: 64-bit Operating System, x64-based processor
- python: 3.8
- PyMuPDF：1.18.14
- fitz：0.0.1.dev2
- Adobe Acrobat Reader DC 2021.005.20048

**Comment by JorjMcKie** [comment, marked as answer]
If you insert "permament" text, there is a lot of freedom with respect to which font etc. to use.
But if you want to be able to later move around the text and thus use FreeText annotations, then (Py-) MuPDF restricts you to (1) the Base-14 fonts Helvetica, Times-Roman and Courier, plus (2) to only one of these within the same annotation.
Whether you will then be able to do this moving with Adobe, PDF-XChange or other viewers, and / or then switch back to PyMuPDF is a different story: this will either not be possible, or force you back to the Base-14 fonts.

**Reply by JorjMcKie** [reply]
The issue you referenced, #549, is about **_permamnent_** text, not annotations.

**Reply by sldzys** [reply]
Thank you so much!  PyMuPDF is currently the most powerful PDF tool in python I think.  I will try some other ways such as FDF to see if I can achieve the function I want.

Thanks again for your quickly replay!

have a good day.👍

**Comment by JorjMcKie** [comment]
This is about the maximum you can do with freetext annotations:
[freetext-annot-lang.zip](https://github.com/pymupdf/PyMuPDF/files/6727996/freetext-annot-lang.zip)
...And not all PDF viewers are even capable to read every annotation generated here.
